c++/src/kj/common.h

```diff
@@ -1416,7 +1416,19 @@ To& downcast(From& from) {
   }
 
 #if !KJ_NO_RTTI
+
+#ifdef __GNUC__
+// GCC 6.3 with -Werror=address will fail to build if To and From are the same type.
```

Comment:
What is the warning it generates? Is this a compiler bug? Can we reference the bug number?

Are we ignoring the warning because it breaks *our* build, or other people's builds? Other people should be unaffected because we mark our headers as system headers. If it's our own build (with -Werror) that's affected, maybe we should adjust our build config to disable this warning if it is buggy?

Reply:
Rebased in a more appropriate fix that removes the superfluous downcast, as discussed in person.